src/wagtailmedia/templates/wagtailmedia/chooser/chooser.html

```diff
@@ -41,7 +41,7 @@
 
     <div class="tab-content">
         <section id="tab-search" class="w-tabs__panel" role="tabpanel" aria-labelledby="tab-label-search">
-            <form action="{% url 'wagtailmedia:chooser' %}" method="GET" role="search" class="media-search search-bar" novalidate data-chooser-modal-search>
+            <form action="{% url 'wagtailmedia:chooser' %}{% if media_type %}{{ media_type }}{% endif %}" method="GET" role="search" class="media-search search-bar" novalidate data-chooser-modal-search>
```

Comment:
this should use `linkurl` (renamed to `chooser_url` to better reflect its purpose) rather than blindly dropping the `media_type` at the end of the generic chooser url.